### PR TITLE
[2041] dfe_analytics module

### PR DESCRIPTION
**Pull request by saliceti**
## Context
New module to configure resources for [dfe-analytics](https://github.com/DFE-Digital/dfe-analytics/) in Google cloud, and provide the required variables on the Azure side.

## Changes proposed in this pull request
The new module can either reuse the existing dataset, and simply provide the variables for the app on AKS, or create the dataset and events table.
It also provides a bach script to configure Google cloud IAM, so the terraform code can be run in a Github action workflow.
The README dscribes the usage.

## Guidance to review
- Review README
- See how it is configured and used [in Apply for QTS](https://github.com/DFE-Digital/apply-for-qualified-teacher-status/pull/2441)
- See successful workflow: https://github.com/DFE-Digital/apply-for-qualified-teacher-status/actions/runs/11457332810
- Access https://apply-for-qts-review-2441-web.test.teacherservices.cloud/ and observe events in bigquery

## Before merging
Remove the TEMP commit

## After merging
Update the module tag in the AFQTS PR

## Checklist

- [x] I have performed a self-review of my code, including formatting and typos
- [x] I have [cleaned the commit history](https://www.annashipman.co.uk/jfdi/good-pull-requests.html)
- [x] I have added the `Devops` label
- [x] I have attached the pull request to the trello card

### CODE REVIEW

**Comment by saliceti** [comment]
[Configure Bigquery in the terraform module](https://trello.com/c/cQwpKIWx/2041-configure-bigquery-in-the-terraform-module)

File: aks/dfe_analytics/README.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+## Taxonomy and policy tag
+The user should have Owner role on the Google project.
+
+- Authenticate: `gcloud auth application-default login`
+- Get projects list: `gcloud projects list`
+- Select project e.g.: `gcloud config set project apply-for-qts-in-england`
+- Get taxonomies list:
+  ```
+  gcloud data-catalog taxonomies list --location=europe-west2 --format="value(name)"`
```
Comment (RMcVelia) [inline aks/dfe_analytics/README.md:116]: rogue ` at the end?

File: aks/dfe_analytics/README.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  ```
+  The path contains the taxonomy id as a number e.g. 5456044749211275650
+- Get policy tags e.g.:
+  ```
+  gcloud data-catalog taxonomies policy-tags list --taxonomy="projects/apply-for-qts-in-england/locations/europe-west2/taxonomies/5456044749211275650" --location="europe-west2" --filter="displayName:hidden" --format="value(name)"`
```
Comment (RMcVelia) [inline aks/dfe_analytics/README.md:121]: rogue ` at the end?

File: aks/dfe_analytics/README.md (hunk 1 above)
Comment (saliceti) [reply aks/dfe_analytics/README.md:116]: done

File: aks/dfe_analytics/README.md (hunk 2 above)
Comment (saliceti) [reply aks/dfe_analytics/README.md:121]: done

File: aks/dfe_analytics/README.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+## Authentication - Command line
+The user should have Owner role on the Google project.
+
+- Run `gcloud auth application-default login`
+- Run terraform
+
+## Authentication - Github actions
+We set up workfload identity federation on the Google side and configure the workflow. The user should have Owner role on the Google project. This is done once per repository.
+
+- Run the `autorise_workflow.sh` located in *aks/dfe_analytics*:
```
[the remark this answers is not in the document]
Comment (saliceti) [reply aks/dfe_analytics/README.md]: done

**Review by RMcVelia — APPROVED** [review]

**Pull request merged**